[Swift] Only apply 'open' to classes and potentially-overridable decls

jrose-apple opened the pull request:
And don't apply it to anything marked `final`.

The test here *would* catch the issue here once the Swift AST verifier is updated to enforce the above conditions, but the AST verifier isn't run after the AST has been manipulated to add `open`.

No intended user-visible effect.

jrose-apple commented:
@swift-ci Please test

dcci commented on source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp:
```diff
@@ -837,9 +837,10 @@ void SwiftASTManipulator::MakeDeclarationsPublic() {
-        if (swift::isa<swift::ClassDecl>(value_decl) ||
-            swift::isa<swift::ClassDecl>(value_decl->getDeclContext())) {
-          access = swift::AccessLevel::Open;
+        if (llvm::isa<swift::ClassDecl>(value_decl) ||
```
Can you add a comment here? The surrounding isn't really much commented but we should try improving this situation moving forward.

jrose-apple committed:
[Swift] Only apply 'open' to classes and potentially-overridable decls

And don't apply it to anything marked 'final'.

The test here /would/ catch the issue here once the Swift AST verifier
is updated to enforce the above conditions, but the AST verifier isn't
run after the AST has been manipulated to add 'open'.

No intended user-visible effect.

jrose-apple commented:
@swift-ci Please test

jrose-apple commented:
@swift-ci Please test

jimingham reviewed (approved):
Makes sense.

jrose-apple commented:
See also apple/swift#15996
